src/asm/asmy.y

```diff
@@ -142,8 +142,9 @@ void	copyrept( void )
 {
 	SLONG	level=1, len, instring=0;
 	char	*src=pCurrentBuffer->pBuffer;
+	char	*bufferEnd = pCurrentBuffer->pBufferStart + pCurrentBuffer->nBufferSize;
```

Comment:
Because `pCurrentBuffer->nBufferSize` is the size from the start of the buffer – not the remaining buffer left (and `src` needs to be set to the current position in the buffer, else it'd try to make a REPT out of somewhere else in the file).